clients/drcachesim/common/trace_entry.h

```diff
@@ -192,29 +192,32 @@ typedef enum {
     TRACE_MARKER_TYPE_CPU_ID,
 
     /**
-     * The marker value contains function id defined by the user in droption.
+     * The marker value contains function id defined by the user in the
+     * -record_function (and -record_heap_value if -record_heap is specified)
+     * option.
      */
     TRACE_MARKER_TYPE_FUNC_ID,
 
     // XXX i#3048: replace return address with callstack information.
     /**
-     * The marker value contains the return address of the function call,
-     * whose id is specified by the closest previous
-     * marker entry.
+     * The marker value contains the return address of the just-entered
+     * function, whose id is specified by the closest previous
+     * #TRACE_MARKER_TYPE_FUNC_ID marker entry.
      */
     TRACE_MARKER_TYPE_FUNC_RETADDR,
 
     /**
-     * The marker value contains one argument value of the function call,
-     * whose id is specified by the closest previous #TRACE_MARKER_TYPE_FUNC_ID
-     * marker entry. The number of such entries for one function invocation is
-     * expected to be equal to the specified arugment in -record_function or
-     * pre-defined functions in -record_heap (if enabled).
+     * The marker value contains one argument value of the just-entered
+     * function, whose id is specified by the closest previous
+     * #TRACE_MARKER_TYPE_FUNC_ID marker entry. The number of such entries
+     * for one function invocation is equal to the specified argument in
+     * -record_function (or pre-defined functions in -record_heap_value if
+     * -record_heap is specified).
      */
     TRACE_MARKER_TYPE_FUNC_ARG,
 
     /**
-     * The marker value contains return value of the function call,
+     * The marker value contains return value of the just-entered function,
```

Comment:
Repeating (!):
grammar: s/return/the return/

Reply:
Done